Framework/DataHandling/src/LoadSampleShape.cpp

```diff
@@ -0,0 +1,203 @@
+  std::vector<uint16_t> triangleIndices;
+  std::vector<V3D> vertices;
+  V3D t1, t2, t3;
+
+  while (readSTLTriangle(file, t1, t2, t3)) {
+    // Add triangle if all 3 vertices are distinct
+    if (!areEqualVertices(t1, t2) && !areEqualVertices(t1, t3) &&
+        !areEqualVertices(t2, t3)) {
+      triangleIndices.push_back(addSTLVertex(t1, vertices));
+      triangleIndices.push_back(addSTLVertex(t2, vertices));
+      triangleIndices.push_back(addSTLVertex(t3, vertices));
+    }
+  }
+  // Use efficient constructor of MeshObject
+  std::unique_ptr<MeshObject> retVal = std::unique_ptr<MeshObject>(
+      new MeshObject(std::move(triangleIndices), std::move(vertices),
+                     Mantid::Kernel::Material()));
+  return retVal;
+}
+
+std::unique_ptr<Geometry::MeshObject> readSTLSolid(std::ifstream &file,
+                                                   std::string &name) {
+  // Read Solid name
+  // We expect line after trimming to be "solid "+name.
+  std::string line;
+  if (getline(file, line)) {
+    boost::trim(line);
+    if (line.size() < 5 || line.substr(0, 5) != "solid") {
+      throw std::runtime_error("Expected start of solid");
+    } else {
+      name = line.substr(6, std::string::npos);
+    }
+    // Read Solid shape
+    return readSTLMeshObject(file);
+  }
+  return nullptr;
+}
+
+} // end anonymous namespace
+
+void LoadSampleShape::init() {
+  auto wsValidator = boost::make_shared<API::InstrumentValidator>();
+  ;
+
+  // input workspace
+  declareProperty(
+      make_unique<WorkspaceProperty<>>("InputWorkspace", "", Direction::Input,
+                                       wsValidator),
+      "The name of the workspace containing the instrument to add the shape");
+
+  // shape file
+  const std::vector<std::string> extensions{".stl"};
+  declareProperty(
+      make_unique<FileProperty>("Filename", "", FileProperty::Load, extensions),
+      "The path name of the file containing the shape");
+
+  // Output workspace
+  declareProperty(make_unique<WorkspaceProperty<>>("OutputWorkspace", "",
+                                                   Direction::Output),
+                  "The name of the workspace that will contain the loaded "
+                  "shape of the sample");
+}
+
+void LoadSampleShape::exec() {
+
+  MatrixWorkspace_const_sptr inputWS = getProperty("InputWorkspace");
+  MatrixWorkspace_sptr outputWS = getProperty("OutputWorkspace");
+
+  if (inputWS != outputWS) {
+    outputWS = inputWS->clone();
+  }
+
+  std::string filename = getProperty("Filename");
+  std::ifstream file(filename.c_str());
+  if (!file) {
+    g_log.error("Unable to open file: " + filename);
+    throw Exception::FileError("Unable to open file: ", filename);
+  }
+
+  std::string solidName = "";
+  boost::shared_ptr<MeshObject> shape = nullptr;
+  try {
+    shape = readSTLSolid(file, solidName);
+  } catch (std::exception &e) {
+    throw Exception::FileError(e.what(), filename);
```

Comment:
What is this try catch intending to do?

This code is at best misleading and potentially dangerous.
How is this catch handling the error?

Reply:
I now realise it is just passing on an exception thrown lower down, so is not needed.